packages/beacon-node/src/chain/lightClient/index.ts

```diff
@@ -501,14 +468,58 @@ export class LightClientServer {
           finalizedHeader,
           syncAggregate,
           finalityBranch: attestedData.finalityBranch,
+          signatureSlot,
         };
-        this.emitter.emit(ChainEvent.lightclientFinalizedUpdate, this.finalized);
+        this.latestForwardedFinalitySlot = this.finalized.attestedHeader.slot;
+        this.emitter.emit(ChainEvent.lightClientFinalityUpdate, this.finalized);
         this.metrics?.lightclientServer.onSyncAggregate.inc({event: "update_latest_finalized_update"});
       }
     }
 
     // Check if this update is better, otherwise ignore
-    await this.maybeStoreNewBestPartialUpdate(syncPeriod, syncAggregate, attestedData);
+    await this.maybeStoreNewBestPartialUpdate(syncPeriod, syncAggregate, signatureSlot, attestedData);
+  }
+
+  private async doGetUpdate(period: number): Promise<altair.LightClientUpdate> {
```

Comment:
Can you call this `getUpdate`? and it can also be public.

Reply:
+1

Reply:
> Can you call this `getUpdate`? and it can also be public.

Doing this will make it a public function that only ever get used within this object as there is no instance where such method call is required by other part of the codebase. Any reason why you will want this?

Also take into consideration that the current use case maps to the query params `?start_period={}&count={}` where even if you need a single update you need to set count to 1.

Reply:
Vaguely, the reason for exposing this was: I like interfaces that expose the underlying functionality / primitives as simply as possible, rather than interfaces with higher-level logic, or logic based on the considerations of an external consumer. Ofc its somewhat subjective so :shrug: 
I think this is idea behind this comment too https://github.com/ChainSafe/lodestar/pull/4365#discussion_r987120494

Reply:
Gotcha. But it can also be _subjectively_ said that the underlying functionality to be exposed is ability to retrieve the updates, given the `start_period` and `count` needed. If the function returns one singular update as being suggested for `getUpdate`. Then usage will be something like:

```
for n in counts {
   lightClientServer.getUpdate(n)
}
```

instead of

```
 lightClientServer.getUpdates(start_period, counts)
``` 

Which I think is clearer.

I left a comment to the comment you pointed to

Reply:
i think @wemeetagain wants to expose both as public fns, which should be fine.

Reply:
> i think @wemeetagain wants to expose both as public fns, which should be fine.

Indeed. And I asked the reasoning behind that, which he also explained and rightly clarified as partly being subjective. I'll rename this in a bit. It does no harm which ever way